src/pubsub/mod_pubsub_db_mnesia.erl

```diff
@@ -259,20 +259,36 @@ remove_all_items(Nidx) ->
 
 -spec add_item(Nidx :: mod_pubsub:nodeIdx(),
                JID :: jid:jid(),
-               ItemId :: mod_pubsub:itemId()) ->
+               ItemId :: mod_pubsub:pubsubItem()) ->
     ok.
-add_item(Nidx, JID, ItemId) ->
+add_item(Nidx, JID, #pubsub_item{itemid = {ItemId, _}} = Item) ->
+    set_item(Item),
     {ok, State} = get_state(Nidx, jid:to_bare(JID), write),
     NewItems = [ItemId | State#pubsub_state.items],
     mnesia:write(State#pubsub_state{ items = NewItems }).
 
 %% ------------------------ Direct #pubsub_item access ------------------------
 
--spec get_items(Nidx :: mod_pubsub:nodeIdx()) ->
+-spec get_items(Nidx :: mod_pubsub:nodeIdx(), gen_pubsub_node:get_item_options()) ->
     {ok, {[mod_pubsub:pubsubItem()], none}}.
-get_items(Nidx) ->
+get_items(Nidx, Opts) ->
     Items = mnesia:match_object(#pubsub_item{itemid = {'_', Nidx}, _ = '_'}),
-    {ok, {lists:reverse(lists:keysort(#pubsub_item.modification, Items)), none}}.
+    Sorted = lists:reverse(lists:keysort(#pubsub_item.modification, Items)),
+    ItemsLimitedByIds = filter_items_by_item_ids(Sorted, maps:get(item_ids, Opts, undefined)),
+    ItemsLimitedByMaxItems = limit_items(ItemsLimitedByIds, maps:get(max_items, Opts, undefined)),
+    {ok, {ItemsLimitedByMaxItems, none}}.
+
+filter_items_by_item_ids(Items, undefined) ->
+    Items;
+filter_items_by_item_ids(Items, ItemIds) ->
```

Comment:
How about adding a guard:
```erlang
filter_items_by_item_ids(Items, ItemIds) when is_list(Items) and is_list(ItemIds) ->
```
??

Reply:
I don't know. On one hand yes, on the other there is dialyzer and function spec for the main function using these helper function and we run dialyzer checks on travis for every PR.